### DISCUSSION TITLE
Weird interaction with ~ and underscores in filters

### DISCUSSION THREAD

**Discussion by DaniDipp** [pocketbase/pocketbase#1231] [Q&A] [answered]
A bug in my app is caused by what seems to be a special rule for how underscores are treated in `~` filter queries. I understand that `~` is used for partial matching, but with an exact match, I expected both `~` and `=` to return the matched result. This worked fine for a while:
![2022-12-11T03-21-11](https://user-images.githubusercontent.com/17002453/206883450-47ff0b30-37f4-470c-b258-1198dac31428.png)
![2022-12-11T03-22-17](https://user-images.githubusercontent.com/17002453/206883456-42f319be-3984-4e18-85ad-9f124117ac17.png)
but then I had a user with an underscore at the end of their name, where this isn't the case:
![2022-12-11T03-23-14](https://user-images.githubusercontent.com/17002453/206883470-c81256cf-2277-415f-a521-757541bcad76.png)
![2022-12-11T03-23-30](https://user-images.githubusercontent.com/17002453/206883473-af00cb41-e151-4226-886d-b998610e748d.png)

How does this syntax work?

**Comment by ganigeorgiev** [comment, marked as answer]
This is a bug with the escaping of the underscore character. The `~` operator converts the expression to a "LIKE" SQL comparison and while we escape the underscore, since it has special meaning for 1 wildcard character, we are missing the `ESCAPE` expression and the query is trying to match the literal `key\_`. I'll fix it sometime later today.

**Reply by ganigeorgiev** [reply]
@DaniDipp _Just to let you know that I've added a fix in `master` and it will be available with the next release (sometime next week or the week following it)._

**Reply by DaniDipp** [reply]
Thanks very much!
